### PR TITLE
Wisp rework

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This PR reworks lavaland wisp lantern, to make it less powerfull validhunting tool, and more usefull in mining.
It makes it give you effects dependant on your current z-level, the best effects are on lavaland and the worst are on the station and away missions. So:
1. If you are on lavaland, it grants you X-ray vision.
2. If you are on the station it grants you meson vision.
3. If you are on any other z-level, it grants you thermal vision.

Additionaly, it grants you night vision if active.

Feel free to discuss in comments.

## Why It's Good For The Game

The wisp is almost useless in mining and very strong in validhunting, and that isn't ok. This PR aims to make it more usefull in mining purposes(in practic frees you from need of wearing meson googles and carrying a flashlight), while making it less strong at validhunting(through, it still will be helpfull and can help you at this purpose.)
image
(less things like this)

## Changelog

:cl: SuperSlayer
balance: Lavaland wisp lantern now has stronger/weaker effect dependant on your Z-level. 
/:cl:

### CODE REVIEW

**Comment by newfren** [comment]
what's the rationale for making it shit in away missions instead of thermal vision like in space

**Comment by SuperSlayer0** [comment]
> what's the rationale for making it shit in away missions instead of thermal vision like in space

I can make them work as in space in away missions if needed.

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mining/lavaland/tendril_loot.dm [hunk 1]
```diff
@@ -258,28 +262,66 @@
...
-/obj/effect/wisp/proc/update_user_sight(mob/user)
-	SIGNAL_HANDLER
-	user.sight |= sight_flags
-	if(!isnull(lighting_alpha))
-		user.lighting_alpha = min(user.lighting_alpha, lighting_alpha)
+/obj/effect/wisp/process()
+	if(!owner)
+		return
+	clear_old_traits()
+	if(SSmapping.level_trait(z, ZTRAIT_MINING))
```
[the remark this answers is not in the document]
Comment (SuperSlayer0) [reply code/modules/mining/lavaland/tendril_loot.dm:298]: Okay, will look how to do this

**Comment by SuperSlayer0** [comment]
Converted to signals, made it work the same on away missions as in space.

**Comment by SuperSlayer0** [comment]
> what's the rationale for making it shit in away missions instead of thermal vision like in space

Fixed, now it works at away missions normaly

File: code/modules/mining/lavaland/tendril_loot.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (SuperSlayer0) [reply code/modules/mining/lavaland/tendril_loot.dm]: Done

**Comment by Bluedino1025** [comment]
How would it work on icebox?

**Comment by SuperSlayer0** [comment]
> How would it work on icebox?

oops

**Comment by Mooshimi** [comment]
If this is still in the works please be sure to test on the different maps and areas

**Pull request closed without merging**